src/v/bytes/tests/iobuf_tests.cc

```diff
@@ -491,4 +491,14 @@ SEASTAR_THREAD_TEST_CASE(trim_front) {
         buf.trim_front(101);
         BOOST_TEST(buf.size_bytes() == 99);
     }
+
+    {
+        iobuf buf;
+        buf.prepend(ss::temporary_buffer<char>(100));
+        auto it = buf.begin();
+        BOOST_TEST(it->size() == 100);
+        buf.trim_front(10);
+        it = buf.begin();
+        BOOST_TEST(it->size() == 90);
```

Comment:
Should this test be a test against the io_fragment and not the iobuf? It seems to me that iobuf::trim_front is not responsible for adhering to any particular internal structure, but this test makes that assumption.

Reply:
We don't have a dedicated io_fragment tests. I think that we can remove this test and leave the ones that I added.